Jenkinsfile

```diff
@@ -17,25 +16,89 @@ node('vetsgov-general-purpose') {
   }
 
   def commonStages = load "vets-website/jenkins/common.groovy"
+  def envUsedCache = [:]
 
   // setupStage
   dockerContainer = commonStages.setup()
 
-  stage('Lint|Security|Unit') {
-    if (params.cmsEnvBuildOverride != 'none') { return }
+  stage('Main') {
+    def contentOnlyBuild = params.cmsEnvBuildOverride != 'none'
+    def assetSource = contentOnlyBuild ? ref : 'local'
 
     try {
       parallel (
         failFast: true,
 
+        buildDev: {
+          if (commonStages.shouldBail()) { return }
+          envName = 'vagovdev'
```

Comment:
In the `buildAll` function, [`envName`](        def envName = VAGOV_BUILDTYPES.get(i)) is plucked from [`VAGOV_BUILDTYPES`](https://github.com/department-of-veterans-affairs/vets-website/blob/fa55666af5dc82d6c84258d33812bf636d14e23d/jenkins/common.groovy#L27), and I'm concerned it's not safe to hard-code the `envName` like this because `VAGOV_BUILDTYPES` is `[BUILD_TYPE_OVERRIDE]` when `DRUPAL_MAPPING.get(params.cmsEnvBuildOverride` is truthy, and it's `['vagovdev', 'vagovstaging', 'vagovprod'] when `DRUPAL_MAPPING.get(params.cmsEnvBuildOverride` is falsy.

Reply:
I think the equivalent here might be to avoid the step (maybe early return?) if `contentOnlyBuild && envName != params.cmsEnvBuildOverride`.

Reply:
> I think the equivalent here might be to avoid the step (maybe early return?) if `contentOnlyBuild && envName != params.cmsEnvBuildOverride`.

@U-DON I think that solution makes sense, but I think I might be confused by the conditional logic in your example.

Is the following description accurate?

>We should only build when it's not a content-only build (`contentOnlyBuild` is `false`) **and** `envName` is `vagovdev`/`vagovstaging`/`vagovprod`, correct?


```
def shouldBuild = !contentOnlyBuild && envName != params.cmsEnvBuildOverride
if (!shouldBuild) { return }
```

Alternatively:

```
def shouldNotBuild = contentOnlyBuild && envName == params.cmsEnvBuildOverride
if (shouldNotBuild) { return }
```

Do you think either of those correct and valuable?

Reply:
The logic I'm going for is to build only if it's not a content-only build or the environment name matches the override. Alternatively, don't build if it's a content-only build and the environment doesn't match the override.

Reply:
Also keeping in mind that we're considering it content-only when that override is present.

Reply:
> The logic I'm going for is to build only if it's not a content-only build or the environment name matches the override. Alternatively, don't build if it's a content-only build and the environment doesn't match the override.
> 
> ```
> def shouldBuild = !contentOnlyBuild || envName == params.cmsEnvBuildOverride
> def shouldNotBuild = contentOnlyBuild && envName != params.cmsEnvBuildOverride
> ```

I updated the PR with your `shouldBuild` logic in https://github.com/department-of-veterans-affairs/vets-website/pull/16965/commits/91ab8896c34e3022d19c1037dee8215949a42dc1.